bambi/plots/effects.py

```diff
@@ -0,0 +1,402 @@
+                )
+                contrast_df["estimate"] = contrast_estimate.comparison[
+                    tuple(contrast.values)
+                ].to_numpy()
+
+            if use_hdi:
+                contrast_df[f"hdi_{lower_bound}%"] = (
+                    contrast_estimate.hdi[tuple(contrast.values)][
+                        f"{response.name}_{response.target}"
+                    ]
+                    .sel(hdi="lower")
+                    .values
+                )
+                contrast_df[f"hdi_{upper_bound}%"] = (
+                    contrast_estimate.hdi[tuple(contrast.values)][
+                        f"{response.name}_{response.target}"
+                    ]
+                    .sel(hdi="higher")
+                    .values
+                )
+            else:
+                contrast_df[f"lower_{lower_bound}%"] = contrast_estimate.hdi[
+                    tuple(contrast.values)
+                ].sel(quantile=lower_bound)
+                contrast_df[f"upper_{upper_bound}%"] = contrast_estimate.hdi[
+                    tuple(contrast.values)
+                ].sel(quantile=upper_bound)
+
+        # if > 2 contrast values, then need the full dataframe to build contrast_df
+        elif len(contrast.values) >= 3:
+            num_rows = comparisons_df.shape[0]
+            contrast_df = comparisons_df.drop(columns=contrast.name)
+            contrast_df.insert(0, "term", contrast.name)
+            contrast_keys = [list(elem) for elem in list(contrast_estimate.comparison.keys())]
+            contrast_df.insert(1, "contrast", contrast_keys * (num_rows // len(contrast.values)))
+
+            estimates = []
+            for val in contrast_estimate.comparison.values():
+                estimates.append(val)
+            contrast_df["estimate"] = np.array(estimates).flatten()
+
+            lower = []
+            upper = []
+            for pair, val in zip(contrast_keys, contrast_estimate.hdi.values()):
+                if use_hdi:
+                    lower.append(
+                        (
+                            contrast_estimate.hdi[tuple(pair)][f"{response.name}_{response.target}"]
+                            .sel(hdi="lower")
+                            .values
+                        )
+                    )
+                    upper.append(
+                        (
+                            contrast_estimate.hdi[tuple(pair)][f"{response.name}_{response.target}"]
+                            .sel(hdi="higher")
+                            .values
+                        )
+                    )
+                else:
+                    lower.append(contrast_estimate.hdi[tuple(pair)].sel(quantile=lower_bound))
+                    upper.append(contrast_estimate.hdi[tuple(pair)].sel(quantile=upper_bound))
+
+            contrast_df[f"lower_{lower_bound}%"] = np.array(lower).flatten()
+            contrast_df[f"upper_{upper_bound}%"] = np.array(upper).flatten()
```

Comment:
Just want to mention I see many f-strings written on the fly. It's not problematic per se. But you may want to consider creating variables to hold these names and then use the variables. Makes it easier to maintain and understand what they represent.

Reply:
I like that tip. Thanks for the feedback!